File: uplift-gradle-plugins/src/main/java/com/github/kjetilv/uplift/plugins/UpliftPingTask.java
package com.github.kjetilv.uplift.plugins;

import org.gradle.work.DisableCachingByDefault;

/** Reports the deployed stack. Read only. */
@DisableCachingByDefault(because = "Pinging is not cacheable")
public abstract class UpliftPingTask extends UpliftTask {

    @Override
    protected void perform() {
        report(null);
    }
}

File: uplift-gradle-plugins/src/main/kotlin/com/github/kjetilv/uplift/plugins/UpliftPingTask.kt
package com.github.kjetilv.uplift.plugins

import org.gradle.process.ExecOperations
import org.gradle.work.DisableCachingByDefault
import javax.inject.Inject

@DisableCachingByDefault(because = "Pinging is not cacheable")
abstract class UpliftPingTask @Inject constructor(execOperations: ExecOperations) : UpliftTask(execOperations) {

    override fun perform() = ping()
}
